book/src/networking.md

```diff
@@ -0,0 +1,84 @@
+# Networking
+
+## Network Link Selection
+
+Ariel OS currently supports two different networking links: Ethernet-over-USB (aka CDC-NCM) and Wi-Fi.
+Boards may support both of them, only one of them, or none of them.
+
+Which link layer is used for networking must be selected at compile time,
+through [laze modules](./build_system.md#laze-modules):
+
+- `usb-ethernet`: Selects Ethernet-over-USB.
+- `wifi-cyw43`: Selects Wi-Fi using the CYW43 chip along an RP2040 MCU (e.g., on the Raspberry Pi Pico W).
+- `wifi-esp`: Selects Wi-Fi on an ESP32 MCU.
+
+When available on the device, one of these module is always selected by default, currently preferring Wi-Fi networking.
+
+Overriding this default selection is possible by explicitly selecting the desired module, as follows:
+
+```sh
+laze build --select usb-ethernet -b rpi-pico-w
+```
+
+## Network Credentials
+
+For Wi-Fi, the network credentials have to be supplied via environment variables:
+
+```sh
+CONFIG_WIFI_NETWORK=<ssid> CONFIG_WIFI_PASSWORD=<pwd> laze build ...
+```
+
+## Using the Networking Link on the Device
+
+### Network Configuration
+
+DHCPv4 is used by default for network configuration, including for IP address allocation.
+
+The [`#[ariel_os::config]` attribute macro][config-attr-macro-rustdoc] is currently used to provide manual network configuration for the device.
+When the `override-network-config` Cargo feature is enabled, DHCP is disabled and the provided configuration is used instead.
+
+> Non-static IPv6 address allocation will be supported in the future.
+
+### Support for Network Protocols
+
+Support for various network protocols can be enabled through [Cargo features listed in the documentation][rustdoc-homepage].
+Most of these use `embassy_net`, which should be used through the [`ariel_os::reexports::embassy_net`][embassy-net-reexport-rustdoc] re-export.
+
+### Obtaining a Network Stack
+
+A network stack can then be obtained using [`ariel_os::network::network_stack()`][network-stack-rustdoc].
```

Comment:
```suggestion
A network stack handle can then be obtained using [`ariel_os::network::network_stack()`][network-stack-rustdoc].
```

Reply:
Maybe add that this then implements the embedded-nal-async traits?

Reply:
Oh, I thought it did. Seems like upstream actually doesn't, and ariel only partially and only in `src/ariel-os-coap/src/udp_nal/mod.rs`-